frontend/src/components/Song.js

```diff
@@ -0,0 +1,149 @@
+import React, { useEffect, useRef, useState } from "react";
+import { Text, Text3D } from "@react-three/drei";
+import { animated, useSpring } from "@react-spring/three";
+import { v4 as uuidv4 } from "uuid";
+
+import data from "../utils";
+const toneArmOnSoundeffect = new Audio("/soundeffects/tonearm_on_sound.mp3");
+toneArmOnSoundeffect.volume = 0.4;
+const vinylSoundeffect = new Audio("/soundeffects/vinyl_soundeffect.mp3");
+vinylSoundeffect.volume = 1;
+vinylSoundeffect.loop = true;
+
+export default function Song({
+  playing,
+  setPlaying,
+  toneArmFinished,
+  soundOn,
+  songIndex,
+}) {
+  const x = -3.2;
+  const y = 1.4;
+  const z = 4;
+
+  const [songs, setSongs] = useState(data()); // List of songs
```

Comment:
If you are not going to use `setSongs`, then we probably don't need a state. Let me know if it works if you just declare a const and not have it be a state for the component.

Reply:
@320834  You're right that we won't need it for now, but I think we might need something similar when we tie in the Spotify API.

_IE_ : If the user builds a queue of songs to listen, then `setSongs` would be needed to update the list of songs (which would be displayed in the frontend).


Thoughts?

Reply:
Yeah understood, but we will add it in when we get to that point.